File: POO/JS/persona.js
class Persona {
    constructor(nombre, edad, carrera, universidad) {
      this.nombre = nombre;
      this.edad = edad;
      this.carrera = carrera;
      this.universidad = universidad;
    }
  
    saludar() {
      return 'Hola, mi nombre es ' + this.nombre + ' y tengo ' + this.edad + ' años.'
    }
  
    cumpleanios() {
      return '¡Felicidades! tienes ' + this.edad + ' años.';
    }
  
    estudiar() {
      return 'Estoy estudiando ' + this.carrera;
    }

    Universidad() {
      return 'Mi universidad es la Catolica de Santa Cruz ' + this.universidad;
    }

    modificar_nombre() {
        this.nombre = prompt("Ingresar tu nuevo nombre")
        while (!this.nombre ||!isNaN(this.nombre)) {
            alert("Nombre invalido, Coloque letras para su nombre porfavor");
            this.nombre = prompt("Ingrese nuevamente el nombre");
        }
        return 'Hola, mi nuevo nombre es ' + this.nombre + ' y tengo ' + this.edad + ' años.'
      }
    
    modificar_cumpleanios() {
        this.edad = prompt ('ingrese su nueva edad');
    while (this.edad  <=0){
        alert("La edad es incorrecta. Ingrese una edad válida");
      ;
    }
    if (this.edad < 0) {
      this.edad = prompt ('ingrese nuevamente su edad')
      return 'Ahora tengo ' + this.edad + ' años ';
    }
      }
    
    modificar_estudiar() {
        this.carrera = prompt("Ingresar su nueva carrera")
        while (!this.carrera || !isNaN(this.carrera)) {
            alert("Carrera invalida, Coloque letras para su carrera");
            this.carrera = prompt("Ingrese nuevamente su carrera");
        }
        return 'Ahora estudio ' + this.carrera;
      }
  
    modificar_Universidad() {
        this.universidad = prompt("Ingresar su nueva universidad")
        while (!this.universidad || !isNaN(this.universidad)) {
            alert("Universidad invalida, coloque letras porfavor");
            this.universidad = prompt("Ingrese nuevamente su universidad");
        }
        return 'Mi nueva universidad es ' + this.universidad;
      }

    eliminar_nombre() {
        delete 'Hola, mi nuevo nombre es ' + this.nombre + ' y tengo ' + this.edad + ' años.'
    }

    eliminar_cumpleanios() {
        delete '¡Felicidades! ahora tienes ' + this.edad + ' años.'
    }

    eliminar_estudiar() {
        delete 'Ahora estudio ' + this.carrera
    }

    eliminar_universidad() {
        delete 'Mi nueva universidad es ' + this.universidad
    }
  }
  
  // Crear una instancia de la clase Persona
  const persona = new Persona('Leandro', 18, 'Ingeniería Industrial', 'UCB');
  
  // Inicializar los elementos HTML
  const saludar = document.getElementById('saludar');
  const edad = document.getElementById('edad');
  const carrera = document.getElementById('carrera');
  const universidad = document.getElementById('universidad');
  
  // Función para saludar
  function botonSaludar() {
    saludar.textContent = persona.saludar();
  }
  
  // Función para cumpleaños
  function botonCumpleanios() {
    edad.textContent = persona.cumpleanios();
  }
  
  // Función para estudiar
  function botonEstudiar() {
    carrera.textContent = persona.estudiar();
  }

  function botonUniversidad() {
    universidad.textContent = persona.Universidad();
  }

  function botonModificar_Saludo() {
    saludar.textContent = persona.modificar_nombre();
  }

  function botonModificar_Cumpleanios() {
    edad.textContent = persona.modificar_cumpleanios();
  }

  function botonModificar_Estudiar() {
    carrera.textContent = persona.modificar_estudiar();
  }

  function botonModificar_Universidad() {
    universidad.textContent = persona.modificar_Universidad();
  }

  function botonEliminar_Saludo(){
    saludar.textContent = persona.eliminar_nombre();
  }

  function botonEliminar_Cumpleanios(){
    edad.textContent = persona.eliminar_cumpleanios();
  }

  function botonEliminar_Estudiar(){
    carrera.textContent = persona.eliminar_estudiar();
  }

  function botonEliminar_Universidad(){
    universidad.textContent = persona.eliminar_universidad();
  }

File: POO/Py/Persona.py
class Persona:
    def __init__(self, nombre, edad, carrera, universidad):
        self.nombre = nombre
        self.edad = edad
        self.carrera = carrera
        self.universidad = universidad

    def saludar(self):
        return f'Hola, mi nombre es {self.nombre} y tengo {self.edad} años.'

    def cumpleanios(self):
        return f'¡Felicidades! tienes {self.edad} años.'

    def estudiar(self):
        return f'Estoy estudiando {self.carrera}'

    def mostrar_universidad(self):
        return f'Mi universidad es la {self.universidad}'

    def modificar_nombre(self):
        while True:
            nuevo_nombre = input("Ingrese su nuevo nombre: ")
            if nuevo_nombre.isalpha():
                self.nombre = nuevo_nombre
                break
            else:
                print("Nombre inválido, por favor ingrese solo letras.")
        return f'Hola, mi nuevo nombre es {self.nombre} y tengo {self.edad} años.'

    def modificar_cumpleanios(self):
        while True:
            try:
                nueva_edad = int(input("Ingrese su nueva edad: "))
                if nueva_edad > 0:
                    self.edad = nueva_edad
                    break
                else:
                    print("La edad es incorrecta. Ingrese una edad válida.")
            except ValueError:
                print("La edad debe ser un número entero.")
        return f'Ahora tengo {self.edad} años.'

    def modificar_estudiar(self):
        while True:
            nueva_carrera = input("Ingrese su nueva carrera: ")
            if nueva_carrera.isalpha():
                self.carrera = nueva_carrera
                break
            else:
                print("Carrera inválida, por favor ingrese solo letras.")
        return f'Ahora estudio {self.carrera}'

    def modificar_universidad(self):
        while True:
            nueva_universidad = input("Ingrese su nueva universidad: ")
            if nueva_universidad.isalpha():
                self.universidad = nueva_universidad
                break
            else:
                print("Universidad inválida, por favor ingrese solo letras.")
        return f'Mi nueva universidad es {self.universidad}'

    def eliminar_nombre(self):
        self.nombre = None
        return "Nombre eliminado."

    def eliminar_cumpleanios(self):
        self.edad = None
        return "Edad eliminada."

    def eliminar_estudiar(self):
        self.carrera = None
        return "Carrera eliminada."

    def eliminar_universidad(self):
        self.universidad = None
        return "Universidad eliminada."


# Crear una instancia de la clase Persona
persona = Persona('Leandro', 18, 'Ingeniería Industrial', 'UCB')

# Menú para interactuar con la persona
while True:
    print("\nMenú:")
    print("1. Saludar")
    print("2. Cumpleaños")
    print("3. Estudiar")
    print("4. Universidad")
    print("5. Modificar nombre")
    print("6. Modificar edad")
    print("7. Modificar carrera")
    print("8. Modificar universidad")
    print("9. Eliminar nombre")
    print("10. Eliminar edad")
    print("11. Eliminar carrera")
    print("12. Eliminar universidad")
    print("13. Salir")
    
    opcion = input("Ingrese una opción: ")
    
    if opcion == "1":
        print(persona.saludar())
    elif opcion == "2":
        print(persona.cumpleanios())
    elif opcion == "3":
        print(persona.estudiar())
    elif opcion == "4":
        print(persona.mostrar_universidad())
    elif opcion == "5":
        print(persona.modificar_nombre())
    elif opcion == "6":
        print(persona.modificar_cumpleanios())
    elif opcion == "7":
        print(persona.modificar_estudiar())
    elif opcion == "8":
        print(persona.modificar_universidad())
    elif opcion == "9":
        print(persona.eliminar_nombre())
    elif opcion == "10":
        print(persona.eliminar_cumpleanios())
    elif opcion == "11":
        print(persona.eliminar_estudiar())
    elif opcion == "12":
        print(persona.eliminar_universidad())
    elif opcion == "13":
        print("Adiós!")
        break
    else:
        print("Opción inválida. Por favor, ingrese una opción válida.")